sfdx-project.schema.json

```diff
@@ -415,616 +149,807 @@
+          ],
+          "type": "object"
+        },
+        {
+          "additionalProperties": false,
+          "properties": {
+            "glob": {
+              "type": "string"
+            },
+            "regexToReplace": {
+              "type": "string"
+            },
+            "replaceWhenEnv": {
+              "items": {
+                "additionalProperties": false,
+                "properties": {
+                  "env": {
+                    "type": "string"
+                  },
+                  "value": {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "env",
+                  "value"
+                ],
+                "type": "object"
               },
-              "replaceWhenEnv": {
-                "type": "array",
-                "items": {
-                  "type": "object",
-                  "properties": {
-                    "env": {
-                      "type": "string"
-                    },
-                    "value": {
-                      "type": "string"
-                    }
+              "type": "array"
+            },
+            "replaceWithFile": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "glob",
+            "regexToReplace",
+            "replaceWithFile"
+          ],
+          "type": "object"
+        },
+        {
+          "additionalProperties": false,
+          "properties": {
+            "allowUnsetEnvVariable": {
+              "type": "boolean"
+            },
+            "glob": {
+              "type": "string"
+            },
+            "regexToReplace": {
+              "type": "string"
+            },
+            "replaceWhenEnv": {
+              "items": {
+                "additionalProperties": false,
+                "properties": {
+                  "env": {
+                    "type": "string"
                   },
-                  "required": [
-                    "env",
-                    "value"
-                  ],
-                  "additionalProperties": false
-                }
-              }
+                  "value": {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "env",
+                  "value"
+                ],
+                "type": "object"
+              },
+              "type": "array"
             },
-            "required": [
-              "glob",
-              "replaceWithEnv",
-              "stringToReplace"
-            ]
+            "replaceWithEnv": {
+              "type": "string"
+            }
           },
-          {
-            "type": "object",
+          "required": [
+            "glob",
+            "regexToReplace",
+            "replaceWithEnv"
+          ],
+          "type": "object"
+        }
+      ]
+    },
+    "alias-147831185-257-1138-147831185-0-1202": {
```

Comment:
@mshanemc this is a funky name. What's its purpose?

Reply:
it's an artifact of the schema generator because the type is intentionally circular (metadataTypes have a `children` property which contains metadataTypes).